control-plane-operator/controllers/hostedcontrolplane/pki/params.go

```diff
@@ -65,6 +65,22 @@ func NewPKIParams(hcp *hyperv1.HostedControlPlane,
 		IngressSubdomain:             globalconfig.IngressDomain(hcp),
 		OwnerRef:                     config.OwnerRefFrom(hcp),
 	}
-	p.NodeInternalAPIServerIP = util.AdvertiseAddressWithDefault(hcp, config.DefaultAdvertiseAddress)
+
+	// If the first serviceCIDR is an IPv4 we need to set the config.DefaultAdvertiseIPv4Address
+	// as fake IP in the node to access the haproxy exposed as kube-api-server-proxy
+	// Even with that, we cannot set more than one AdvertiseAddress so both
+	// are not supported at the same time.
+	// Check this for more info: https://github.com/kubernetes/enhancements/issues/2438
+	ipv4, err := util.IsIPv4(p.ServiceCIDR)
+	if err != nil {
+		fmt.Printf("error checking the ServiceNetworkCIDRs: %v", err)
```

Comment:
why is this error being ignored? if it's intentional please use the logger to print

Reply:
Here 3 scenarios could happen:

- IsIPv4 - This is an IPv4 and we set the proper default address
- !IsIPv4 - This is an IPv6 and we set the proper default address
- Error on validation - This could happen because the ServiceNetwork is not validated, which is something that apimachinery will not allow. This should not happen or at least I don't see how this scenario could happen. In this case it will default the IPv6 address because booleans are initialized as false by default.

Reply:
We should be using the logger if possible to print out the error message... not fmt.Printf

Reply:
I've decided to just avoid the error message, because the api machinery will hit the issue before and it will report the error early.